thijsvanloef/palworld-server-docker
# Changed backup so only old backup files are deleted

## Context 

* Currently anything older than `OLD_BACKUP_DAYS` in the backup folder is deleted including directories or non backup files
* Changes here ensure only only backup files are deleted

## Choices

* Added additional flag to delete only files
* Added additional flag so only backups which have not had the names modified are deleted
* Added checks to verify OLD_BACKUP_DAYS is defined and is an integer.

## Test instructions

1. Start container with `DELETE_OLD_BACKUPS=true`
2. Enter container: `docker exec -it palworld-server bash`
3. Create at least 4 backups
4. Go to the backup folder: `cd /palworld/backups`
5. Change of the modified date the backups: `find /palworld/backups/ -exec touch -d "9001 days ago" {} \;`
6. Create 2 directories: `mkdir keep empty`
7. Move a backup to keep: `mv palworld-save-2024-01-.... keep/`
8. Rename one backup: `mv palworld-save-2024-01-.... important_palworld-save-2024-01-....`
9. Run backup
10. Verify that empty and keep directories still exists
11. Verify the backup in keep still exists
12. Verify that the backup which you renamed still exists
13. Start the container with `DELETE_OLD_BACKUPS=` and run a backup.
14. Verify backup command outputs `Unable to deleted old backups, OLD_BACKUP_DAYS is empty.`
15. Start the container with `DELETE_OLD_BACKUPS=30b` and run a backup.
14. Verify backup command outputs `Unable to deleted old backups, OLD_BACKUP_DAYS is not an integer. OLD_BACKUP_DAYS=30b`

## Checklist before requesting a review
